src/constructs/mainPipeline.ts

```diff
@@ -61,7 +61,10 @@ export class MainPipeline extends Construct {
                 ],
                 primaryOutputDirectory: props.cdkOutputDirectory,
             }),
-            crossAccountKeys: true,
+            codePipeline: new Pipeline(this, Stack.of(this).stackName, {
```

Comment:
I didn't test it, but I think providing custom Pipeline here will break things:

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.pipelines-readme.html#using-an-existing-aws-codepipeline

> Note that if you provide an existing pipeline, you cannot provide values for pipelineName, crossAccountKeys, reuseCrossRegionSupportStacks, or role because those values are passed in directly to the underlying codepipeline.Pipeline.

Also: https://github.com/aws/aws-cdk/blob/fdf53ba80429d598c54fce0d53d622286d85b6d3/packages/aws-cdk-lib/pipelines/lib/codepipeline/codepipeline.ts#L443

As per docs, using a custom Pipeline requires a bit more config here.

I'm not against it, especially since using a custom Pipeline would enable switching to CodePipeline v2, so feel free to do it. However, note that in sourcecode it says the `restartExecutionOnUpdate` is enabled for a reason:

>         // This is necessary to make self-mutation work (deployments are guaranteed
>        // to happen only after the builds of the latest pipeline definition).